wavedump/src/analyze-waveforms

```diff
@@ -90,134 +83,284 @@ def get_bins(x):
-    parser.add_argument('-u','--upload', default=False, action='store_true', help='upload results to the database')
-    args = parser.parse_args()
+    parser.add_argument('--chunks', default=10000, type=int, help='number of waveforms to process at a time')
+    parser.add_argument('--pdf', default=False, action='store_true', help='prints pdfs')
+    parser.add_argument('--IT', default=300, type=float, help='SPE integration length')
+    parser.add_argument('--s', default=50, type=float, help='start time of the SPE integration')
+    parser.add_argument('--active', default=None, help='Only take data from a single channel. If not specified, all channels are analyzed.')
 
-    data = {}
-    data['ch_511'] = []
-    data['ch_511_rise_time'] = []
-    data['ch_511_fall_time'] = []
+    args = parser.parse_args()
 
     charge = {}
-    with h5py.File(args.filename) as f:
-        data['barcode'] = int(f.attrs['barcode'])
-        data['voltage'] = int(f.attrs['voltage'])
-        data['git_sha1'] = f.attrs['git_sha1'].decode("UTF-8")
-        data['git_dirty'] = f.attrs['git_dirty'].decode("UTF-8")
-
-        x = np.linspace(0,f.attrs['record_length']*1e3/f.attrs['drs4_frequency'],f.attrs['record_length'])
-        for i in range(16):
-            channel = 'ch%i' % i
-
-            if channel not in f:
-                print("failed to find channel %i" % i)
-                continue
-
-            # Skip the first 10 events since that's after we transition from
-            # transparent mode to output mode
-            y = f[channel][10:]*2/0xfff
-
-            # Remove baseline
-            y -= np.median(y[:,:100],axis=-1)[:,np.newaxis]
-
-            a, b = get_window(x,y,left=20,right=400)
-            
-            charge[channel] = integrate(x,y, a, b)
-
-            # FIXME: Should only get average pulse from events near 511 peak
-            if 'avg_pulse_y' in data:
-                data['avg_pulse_y'] += y.mean(axis=0)
-                data['avg_pulse_count'] += 1
-            else:
-                data['avg_pulse_x'] = x
-                data['avg_pulse_y'] = y.mean(axis=0)
-                data['avg_pulse_count'] = 1
-
-            data['ch_511_rise_time'].append(np.median(get_rise_time(x, y)))
-            data['ch_511_fall_time'].append(np.median(get_fall_time(x, y)))
-            
-            if args.plot:
-                plt.subplot(4,4,i+1)
-                plt.plot(x,y[:100].T)
-                plt.xlabel("Time (ns)")
-                plt.ylabel("Voltage (V)")
-                plt.title(channel)
-                plt.axvline(x[a])
-                plt.axvline(x[b])
-
-    data['avg_pulse_y'] /= data['avg_pulse_count']
-    data['avg_pulse_x'] = list(map(float,data['avg_pulse_x']))
-    data['avg_pulse_y'] = list(map(float,data['avg_pulse_y']))
-
-    if args.plot:
-        plt.figure()
-
+    f_charge = {}
+    # `average` and `std` just used for testing.
+    # average = {}
+    # std = {}
```

Comment:
for stuff just used for testing, I think it's best if you create a branch with these, but delete them from the main code